ygot/schema_test.go

```diff
@@ -15,63 +15,88 @@
-			0x74, 0x12, 0x1e, 0xe9, 0xf6, 0x03, 0xcf, 0xd4, 0xfe, 0xba, 0xc1, 0x51, 0xc4, 0xc0, 0xa6, 0x72,
-			0x3d, 0xe8, 0x6c, 0x6c, 0x4b, 0x0f, 0xca, 0x9b, 0xac, 0x7c, 0x5e, 0xa5, 0x92, 0xfa, 0x67, 0x16,
-			0x45, 0x25, 0x61, 0xeb, 0xf6, 0x8c, 0x53, 0xca, 0xf8, 0x40, 0x9d, 0x56, 0x91, 0x3e, 0xcb, 0xb0,
-			0xdc, 0x53, 0x7f, 0xb4, 0xc1, 0x7e, 0xae, 0xf3, 0x12, 0x95, 0xcd, 0x54, 0xae, 0x35, 0x28, 0xae,
-			0xbc, 0x01, 0x00, 0x00, 0xff, 0xff, 0x01, 0x00, 0x00, 0xff, 0xff, 0x85, 0x8c, 0xf4, 0x9f, 0x6d,
-			0x01, 0x00, 0x00,
+			0x1f, 0x8b, 0x08, 0x00, 0x00, 0x00, 0x00, 0x00, 0x02, 0xff, 0x74, 0x90, 0xb1, 0xce, 0xc2, 0x30,
+			0x0c, 0x84, 0xf7, 0x3e, 0x85, 0xe5, 0xb9, 0x52, 0xff, 0xb9, 0xdb, 0x2f, 0xd8, 0x90, 0x78, 0x87,
+			0xa8, 0x4d, 0x69, 0x24, 0x62, 0xa3, 0xe2, 0x4e, 0x28, 0xef, 0x8e, 0x4a, 0x50, 0xc0, 0xa9, 0xd9,
+			0xe2, 0xbb, 0x73, 0xee, 0x93, 0x1f, 0x0d, 0x00, 0x00, 0x9e, 0x5d, 0xf4, 0xd8, 0x03, 0x62, 0x9b,
+			0xe7, 0x53, 0xa0, 0x11, 0x7b, 0xf8, 0x7b, 0x8f, 0x07, 0xa6, 0x29, 0x5c, 0xbe, 0x84, 0x63, 0x58,
+			0xb0, 0x87, 0xbc, 0xfc, 0x12, 0x06, 0x26, 0x71, 0x81, 0xbc, 0x96, 0xd5, 0xdf, 0x9f, 0x48, 0xab,
+			0x03, 0xba, 0xac, 0xc8, 0x75, 0x69, 0x31, 0xea, 0xf2, 0x62, 0x4c, 0xcc, 0xa6, 0xa1, 0x30, 0xb6,
+			0x50, 0x6b, 0x47, 0x6c, 0x10, 0x03, 0x68, 0xe7, 0x27, 0xa5, 0xa4, 0x0a, 0xf8, 0x9f, 0x88, 0xc5,
+			0x49, 0x60, 0xb2, 0xb9, 0xef, 0xc3, 0xec, 0xa3, 0xbb, 0x39, 0x99, 0x37, 0xc0, 0x2e, 0xf2, 0xb8,
+			0x5e, 0x7d, 0xf7, 0xeb, 0x5e, 0x79, 0x45, 0x96, 0x75, 0x10, 0xda, 0x9d, 0xb6, 0xb1, 0xa9, 0xf2,
+			0x2b, 0x35, 0xe9, 0x09, 0x00, 0x00, 0xff, 0xff, 0x01, 0x00, 0x00, 0xff, 0xff, 0x03, 0x33, 0xd7,
+			0x76, 0xf0, 0x01, 0x00, 0x00,
 		},
 		want: map[string]*yang.Entry{
 			"container": containerElement,
 		},
+	}, {
+		name:    "bad gzip data",
+		in:      []byte("I am not a valid gzip!"),
+		wantErr: "gzip: invalid header",
+	}, {
+		name: "bad JSON document",
+		in: []byte{
+			0x1f, 0x8b, 0x08, 0x00, 0x00, 0x00, 0x00, 0x00, 0x02, 0xff, 0xaa, 0x56, 0xca, 0xcc, 0x2b, 0x4b,
+			0xcc, 0xc9, 0x4c, 0xd1, 0xcd, 0x2a, 0xce, 0xcf, 0x53, 0xaa, 0x05, 0x00, 0x00, 0x00, 0xff, 0xff,
+			0x01, 0x00, 0x00, 0xff, 0xff, 0x74, 0x4e, 0x31, 0xbb, 0x10, 0x00, 0x00, 0x00,
+		},
+		wantErr: "invalid character '}' after object key",
+	}, {
+		name:    "empty input",
+		in:      []byte{},
+		wantErr: "EOF",
 	}}
 
 	for _, tt := range tests {
 		got, err := GzipToSchema(tt.in)
 		if err != nil {
-			if !tt.wantErr {
-				t.Errorf("%s: GzipToSchema(%v): got unexpected error: %v\n", tt.name, tt.in, err)
+			if err.Error() != tt.wantErr {
+				t.Errorf("%s: GzipToSchema(%v): got unexpected error, got: %v, want: %v\n", tt.name, tt.in, err, tt.wantErr)
 			}
 			continue
 		}
 
 		if !reflect.DeepEqual(got, tt.want) {
-			t.Errorf("%s: GzipToSchema(%v): did not get expected output, got: %#v, want: %#v", tt.name, tt.in, got, tt.want)
+			gotj, _ := json.MarshalIndent(got, "", strings.Repeat(" ", 4))
+			wantj, _ := json.MarshalIndent(tt.want, "", strings.Repeat(" ", 4))
```

Comment:
may be worthwhile to check for err here - if want is malformed if will lead to a hard to debug error.

Reply:
True, done.